app/src/features/surveys/view/survey-spatial/components/animal/SurveySpatialAnimalTable.tsx

```diff
@@ -9,80 +12,81 @@ import { ScientificNameTypography } from 'features/surveys/animals/components/Sc
 import { useSurveyContext } from 'hooks/useContext';
 import { useCritterbaseApi } from 'hooks/useCritterbaseApi';
 import useDataLoader from 'hooks/useDataLoader';
+import { useEffect } from 'react';
 
-// Set height so the skeleton loader matches table rows
 const rowHeight = 52;
 
-/**
- * Interface defining the structure of animal data used in the table.
- */
-interface IAnimalData {
+interface IAnimalRow {
   id: number;
-  animal_id: string;
+  animal_id: string | null;
   scientificName: string;
+  sex: string | null;
+  status: string;
 }
 
-/**
- * Props interface for SurveySpatialAnimalTable component.
- */
 interface ISurveyDataAnimalTableProps {
   isLoading: boolean;
 }
 
 /**
- * Component for displaying animal data in a table, fetching data via context and API hooks.
- * Renders a table with animal nicknames and scientific names, with loading skeleton when data is loading.
+ * Returns table of animals in the Survey
+ *
+ * @param {ISurveyDataAnimalTableProps} props
+ * @returns {*}
  */
 export const SurveySpatialAnimalTable = (props: ISurveyDataAnimalTableProps) => {
-  const surveyContext = useSurveyContext();
-  const critterbaseApi = useCritterbaseApi();
+  const { isLoading } = props;
 
-  // Fetch critter data loader from survey context
-  const animals = surveyContext.critterDataLoader.data ?? [];
+  const { critterDataLoader } = useSurveyContext();
+  const critterbaseApi = useCritterbaseApi();
+  const animals = critterDataLoader.data ?? [];
 
-  // DataLoader to fetch detailed critter data based on IDs from context
   const animalsDataLoader = useDataLoader(() =>
-    critterbaseApi.critters.getMultipleCrittersByIds(animals.map((animal) => animal.critterbase_critter_id))
+    critterbaseApi.critters.getMultipleCrittersByIds(
+      animals.map(({ critterbase_critter_id }) => critterbase_critter_id)
+    )
   );
 
-  // Load data if animals data is available
-  if (animals.length) {
-    animalsDataLoader.load();
-  }
+  useEffect(() => {
+    if (animals.length) animalsDataLoader.load();
+  }, [animals]);
```

Comment:
React is warning about a missing dependency.

There are a few ways to see the warnings. The easiest is to look at the `app` logs, which you can see if you run `make app-log`.

Additionally, if you have the `eslint` and `prettier` extensions installed in vscode, it will actually highlight them in the editor.

There are some cases where we intentionally omit dependencies from the array because it leads to some infinite loop. But generally we want to try and add all dependencies when possible as it helps prevent unexpected behaviours around when things should be rendering but arent, etc.